src/blog/jwt-identification-and-hot-streams/index.md

```diff
@@ -0,0 +1,172 @@
+      # The #deserialize_identifiers method is a counterpart to #serialize_identifiers
+      # and is responsible for converting strings back to objects
+      identifiers = deserialize_identifiers(JSON.parse(decoded_token["ext"]))
+
+      # Populate identifiers
+      identifiers.each do |k, v|
+        self.public_send("#{k}=", v)
+      end
+    rescue JWT::DecodeError
+      reject_unauthorized_connection
+    end
+  end
+end
+```
+
+Doesn't look like a simple solution 😕 If you'll continue staring at this piece of code for a while, you may notice that the implementation has nothing specific to a particular application: we retrieve a token, verify it and extract identifiers. If so, why can't we re-implement this in Go? We can!
+
+This is how AnyCable Go PRO JWT identification works:
+
+- A client passes a token along the connection request (via a query string or a header).
+- Go server retrieves a header, verifies its signature and TTL, and rejects the connection if token is invalid.
+- If token is valid, we extract identifiers, store them in the connection metadata and finalize the handshake (by sending a `{"type":"welcome"}` message).
+
+And we do not perform any RPC calls! Let's see how it affects the connection time:
+
+<figure class="blog--figure">
+<video loop="loop" muted="muted" autoplay="autoplay" playsinline="true" width="1221" height="800" class="blog--media">
+  <source src="./connect-benchmark.h264.mp4" ttype="video/mp4; codecs=avc1.4D401E,mp4a.40.2">
+</video>
+</figure>
+
+More than 2x faster! What's more important, we reduce the stress on the RPC server. That could be especially useful during _connection avalanches_.
+
+> JWT identification allows you to standardize the authentication flow for WebSockets, saves from cross-site WebSocket hijacking and brings a performance boost!
+
+And that's what you can get with AnyCable PRO and a little companion gem, [anycable-rails-jwt][], which adds helpers to generate tokens.
+
+The only question left is how to handle tokens expiration gracefully? AnyCable Go uses a specific disconnect reason to distinguish expiration from unauthorized access (sends a `{"type":"disconnect","reason":"token_expired"}` message). You can use it to refresh the token. We plan to add a built-in refresh functionality to our [anycable-client][] library soon. Stay tuned!
+
+## Adding some "hot wires" to the equation
+
+We've collected more than a hundred responses from Action Cable / AnyCable users around the world, and found that a good portion of them build applications on top of [Hotwire][hotwire] or [Stimulus Reflex][sr] (+ [CableReady][cr]). Which is pretty cool: [_frontendless_ Rails frontend][frontendless-rails] is getting more and more traction!
+
+What can AnyCable give to these users? Let's take a look at how Turbo Streams work with Rails.
+
+We have the [turbo-rails][] gem, which integrate Action Cable with Hotwire. All you need is to drop a helper into your template to start consuming streams:
```

Comment:
We'll start with the [turbo-rails][] gem, which integrates Action Cable with Hotwire. All you need to do is to drop a helper into your template in order to start consuming streams:

Reply:
Let's start with the [turbo-rails][] gem, which integrates Action Cable with Hotwire. You just need to drop a helper into your template to start consuming streams: